classes/SlackConnector.class.js

```diff
@@ -12,23 +13,33 @@ function SlackConnector(config){
     this.text = null,
     this.response = '';
     this.slack = null;
+    this.activeUsersArray = [];
+    this.slashbot = null;
 
 }
 
 SlackConnector.prototype = {
 	init: function(slashbot){
-		
+		var that = this;
+		this.slashbot = slashbot;
 		console.log("Initializing with SlackConnector...");
 		var slack = new Slack(this.token, this.autoReconnect, this.autoMark);
 		slack.on('open', function() {
-			var channels = ["#test-bot"],
+			var channels = [that.config.channel],
 			    groups = [],
 			    unreads = slack.getUnreadCount(),
 			    key;
+
 			for (key in slack.channels) {
 				if (slack.channels[key].is_member) {
 					channels.push('#' + slack.channels[key].name);
 					console.log('pushing: #' + slack.channels[key].name);
+					for(var i = 0; i < slack.channels[key].members.length; i++){
+						if(slack.getUserByID(slack.channels[key].members[i]).presence == 'active'){
+							that.activeUsersArray.push(slack.getUserByID(slack.channels[key].members[i]).name);							
+						}
+					}
+					slashbot.registerPlayers(that.activeUsersArray);
```

Comment:
Umm - I'd inject the function to register users rather than the slashbot object to register users to. Maybe?